### ISSUE TITLE
google_compute_region_backend_service is missing port_name field

### ISSUE THREAD

**Issue by igorwwwwwwwwwwwwwwwwwwww** [terraform-providers/terraform-provider-google#5551]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

TLDR: The `google_compute_region_backend_service` resource is missing the `port_name` field, which appears to be defaulting to `"http"` at the moment.

This breaks configurations where `protocol` is `"HTTPS"` `load_balancing_scheme` is `"INTERNAL_MANAGED"`, and a named port other than `"http"` is targeted.

For the time being we have set this manually via `gcloud` (with `lifecycle.ignore_changes = all` to prevent terraform from resetting it):

```sh
$ gcloud backend-services update default --port-name custom
```

I've taken a stab at a patch, but got stuck. The header in [`resource_compute_region_backend_service.go`](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_region_backend_service.go) claims the file is auto-generated, but I was not able to find any source [in magic modules](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/resources). I'd be happy to open a PR from [my branch](https://github.com/terraform-providers/terraform-provider-google/compare/master...igorwwwwwwwwwwwwwwwwwwww:region-backend-service-named-port) and discuss further in the PR (let me know).

It's worth noting that there appear to be several other fields that are present in `google_compute_backend_service` but missing from `google_compute_region_backend_service`.

### New or Affected Resource(s)

* `google_compute_region_backend_service`

### Potential Terraform Configuration

```tf
resource "google_compute_region_backend_service" "default" {
  name                  = "default"
  protocol              = "HTTPS"
  load_balancing_scheme = "INTERNAL_MANAGED"

  # this is the missing field that we need
  # port_name             = "custom"

  backend {
    balancing_mode = "UTILIZATION"
    group          = google_compute_instance_group.default.self_link
  }

  health_checks = [google_compute_health_check.http.self_link]
}

resource "google_compute_instance_group" "default" {
  ...

  named_port {
    name = "custom"
    port = "8080"
  }
}
```

### References

* https://github.com/terraform-providers/terraform-provider-google/issues/4935

cc @craigfurman

**Comment by danawillow** [comment]
For context, many fields were omitted from the region backend service resource because they couldn't actually be set on it, and we prefer to keep the resource only with fields that a user can actually set (even if they're present in the API). However, things change over time and it's possible that something that didn't used to work for a resource now does. For this specific one, the [docs](https://cloud.google.com/compute/docs/reference/rest/v1/regionBackendServices) say:
> Must be omitted when the loadBalancingScheme is INTERNAL

My recollection is that region backend services used to only support a loadBalancingScheme of INTERNAL, which would be why that field wouldn't work. Now that it can be INTERNAL_SELF_MANAGED, it's probably worth adding.

The reason you weren't able to find the source in magic modules is that it's fully generated (the ones in third_party are mostly handwritten, and so we just copy them directly from there to here). Take a look at the README at https://github.com/GoogleCloudPlatform/magic-modules and the codelab at https://codelabs.developers.google.com/codelabs/magic-modules/index.html for some information on how MM works. These changes will be smaller than the one in the codelab since it's just adding a field on an existing resource rather than adding a whole new resource.

If you're still interested in adding the missing fields, let me know and I'll assign this issue to you!

**Comment by Alfablos** [comment]
Hi guys,
we're pushing a little bit here: [google backend_service to container_node_pool and named ports #1480](https://github.com/terraform-providers/terraform-provider-google/issues/1480)

There you'll find a solution (a hack, actually) I've found.
Hope it helps

**Comment by quentinleclerc** [comment]
Hello, 
@Alfablos it seems that it's not the same problem. Here the issue is that google_compute_region_backend_service is missing a field. It can not be solved with the workaround from container node_pool not containing named_ports!

Any news on this particular issue? I see that @igorwwwwwwwwwwwwwwwwwwww made a commit? @danawillow 

Thanks!

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
@quentinleclerc Unfortunately I was not able to submit the patch due to the CLA requirement on the magic-modules repo. Somebody else is going to have to do that.

**Comment by danawillow** [comment]
Looks like there's already an open pr: https://github.com/GoogleCloudPlatform/magic-modules/pull/3265
